pyccl/halos/profiles_2pt.py

```diff
@@ -128,9 +128,14 @@ def fourier_2pt(self, prof, cosmo, k, M, a,
         """
         if not isinstance(prof, HaloProfileHOD):
             raise TypeError("prof must be of type `HaloProfileHOD`")
-
         if prof2 is not None:
-            if prof2 is not prof:
-                raise ValueError("prof2 must be the same as prof")
+            if not isinstance(prof2, HaloProfileHOD):
+                raise TypeError("prof2 must be of type "
+                                "`HaloProfileHOD` or None")
+        else:
+            prof2 = prof
+
```

Comment:
I think the 6 preceding lines are pointless, right? We don't really use prof2 for anything other than to check if it's `prof`, so why not just include the two lines right after this comment within the `if prof2 is not None:` bracket?

Reply:
All these kinds of type checks are pointless. Python relies on duck-typing a lot, there is no need to check everything, every time. If the type isn't correct the code will exit and will almost always have a useful error.
There's a lot of stuff to remove in `halo_model.py` too but I suggest to leave it for the v3 branch.

Reply:
(I have fixed this particular one)